### PR TITLE
client settable server address Fixes #1667

### LINKED ISSUE #1667: client should have settable server address
to allow mobile app to reuse the client codes

### PR DESCRIPTION
**Pull request by hamidzr**
changes how the server is addressed from the client side so that the client can be served separately from the server (eg, a mobile application)

### CODE REVIEW

File: src/client/netsblox-dev.html [hunk 1]
```diff
@@ -25,6 +25,10 @@
         <script type="text/javascript" src="Snap--Build-Your-Own-Blocks/FileSaver.min.js"></script>
 
         <!-- Snap overrides -->
+        <script type="text/javascript" >
+            const SERVER_ADDRESS = 'https://editor.netsblox.org',
```
Comment (brollb) [inline src/client/netsblox-dev.html]: We should not to mix deployment configuration information into the code base (though I know there is one in the server CORS checking atm). This could be configured by an environment variable (added to the html template by the server) with the netsblox deployment as a default though

File: src/client/netsblox-dev.html [hunk 2]
```diff
@@ -25,6 +25,10 @@
...
+                SERVER_HOST = SERVER_ADDRESS.replace(/^.*\/\//, '');
```
Comment (brollb) [inline src/client/netsblox-dev.html]: Also, it might make more sense to call the url (with the protocol) `SERVER_URL` and use `SERVER_ADDRESS` for the address without the protocol (rather than `SERVER_HOST`)

Comment (hamidzr) [reply src/client/netsblox-dev.html]: we could do that. I named them after location.host and location.origin

**Comment by hamidzr** [comment]
blocking the mobile development

File: src/server/server.js [hunk 3]
```diff
@@ -130,11 +130,12 @@ Server.prototype.configureRoutes = function() {
             return;
         }
         
-        var baseUrl = `https://${req.get('host')}`,
+        var baseUrl = process.env.SERVER_URL || 'https://editor.netsblox.org',
```
[the remark this answers is not in the document]
Comment (hamidzr) [reply src/server/server.js]: that makes sense, good point

**Comment by hamidzr** [comment]
Does the diff coverage ignore clientside tests?

**Comment by hamidzr** [comment]
resolved the merge conflicts from `test/unit`.

File: src/server/server.js [hunk 4]
```diff
@@ -130,11 +130,12 @@ Server.prototype.configureRoutes = function() {
             return;
         }
         
-        var baseUrl = `https://${req.get('host')}`,
+        var baseUrl = process.env.SERVER_URL,
```
Comment (brollb) [inline src/server/server.js]: Why is this changed?

**Comment by brollb** [comment]
Not sure why it is failing on travis... It is passing locally.
